example/src/main/java/com/stripe/example/activity/PaymentIntentActivity.java

```diff
@@ -0,0 +1,255 @@
+                        new Action1<PaymentIntent>() {
+                            @Override
+                            public void call(PaymentIntent paymentIntent) {
+                                mPaymentIntentValue.setText(paymentIntent.toJson().toString());
+                            }
+                        },
+                        new Action1<Throwable>() {
+                            @Override
+                            public void call(Throwable throwable) {
+                                Log.e(TAG, throwable.toString());
+                            }
+                        }
+                );
+        mCompositeSubscription.add(subscription);
+    }
+
+    void confirmPaymentIntent(final String clientSecret, Card card) {
+        final SourceParams sourceParams = SourceParams.createCardParams(card);
+        final Observable<PaymentIntent> paymentIntentObservable =
+                Observable.fromCallable(
+                        new Callable<PaymentIntent>() {
+                            @Override
+                            public PaymentIntent call() throws Exception {
+                                PaymentIntentParams paymentIntentParams =
+                                        PaymentIntentParams.createConfirmPaymentIntentWithSourceData(
+                                                sourceParams,
+                                                clientSecret,
+                                                RETURN_SCHEMA);
+                                return mStripe.confirmPaymentIntentSynchronous(
+                                        paymentIntentParams,
+                                        PaymentConfiguration.getInstance().getPublishableKey());
+                            }
+                        });
+        Subscription subscription = paymentIntentObservable
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnSubscribe(new Action0() {
+                    @Override
+                    public void call() {
+                        mProgressDialogController.setMessageResource(R.string.confirming_payment_intent);
+                        mProgressDialogController.startProgress();
+                    }
+                })
+                .doOnUnsubscribe(new Action0() {
+                    @Override
+                    public void call() {
+                        mProgressDialogController.finishProgress();
+                    }
+                })
+                .subscribe(
+                        // to the result of creating a 3DS Source
+                        new Action1<PaymentIntent>() {
+                            @Override
+                            public void call(PaymentIntent paymentIntent) {
+                                mPaymentIntentValue.setText(paymentIntent.toString());
+                                if (paymentIntent.getStatus().equals("requires_source_action")) {
```

Comment:
`getStatus()` is marked as `@NonNull`, but I don't actually see that being enforced by `PaymentIntent`

https://github.com/stripe/stripe-android/blob/67c34a31bcd48a71e4451bb4e713c673a7b51179/stripe/src/main/java/com/stripe/android/model/PaymentIntent.java#L232

Same for the other `@NonNull` properties.

Reply:
so we don't really have precedent for doing that. it doesn't actually look like any of the model objects have their fields annotated. I'll go ahead and remove them annotations on these. 

(basically, I'm kind of conflicted - my thinking is annotation are nice, especially for Kotlin users but also I don't want to make more assumptions client side than necessary and enforcing them would require a client side change if that ever changed server-side)

Reply:
That's fair. We definitely do have a precedent for enforcing a minimal set of required fields in the iOS SDK. If they're not there, the object doesn't deserialize (returning `nil` instead), so we actually can annotate some stuff as non-null.